### PR TITLE
Popup

### PR DESCRIPTION
**Pull request by thasnim-cm**

### CODE REVIEW

File: src/components/common/promotioncard.tsx [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+import CheckCircle from "../../assets/check_circle.svg";
+import CustomCard from "./customcard";
+
+type Props = {
+  promotion: IPromotion;
+};
+
+const PromotionCard: FC<Props> = ({ promotion }) => {
+  return (
+    <CustomCard
```
Comment (raj-kumar-litmus) [inline src/components/common/promotioncard.tsx:13]: Children is not passed like this. Try this and see if it works for you.

 ```
<CustomCard
      className="my-2 promotion-card w-[310px] sm:w-[390px] text-white bg-[#30343B] shadow-[0px_2px_6px_#00000033] rounded-lg"
      header={
        <div className="flex items-center px-4 pt-4 pb-2 border-solid border-b border-b-[#383F47]">
          <CustomImage src={CheckCircle} alt="promotion" />
          <span className="text-[#faf9f6] text-sm pl-4 uppercase font-bold text-base">
            {promotion.claimableId}
          </span>
        </div>
      }>
      <div className="flex p-0 justify-between">
          <InfoField title="Promotion" data={promotion.promotionId} />
          <InfoField
            title="Discount Amount"
            data={Number(promotion.discAmt).toFixed(2)}
          />
        </div>
 </CustomCard>
```

Comment (thasnim-cm) [reply src/components/common/promotioncard.tsx:13]: Yes.. it works. I will update

File: src/components/common/promotioncard.tsx [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+const InfoField = ({
+  title,
+  data,
+}: {
+  title: string;
+  data: string | number;
+}) => {
+  return (
+    <div className="block">
```
Comment (raj-kumar-litmus) [inline src/components/common/promotioncard.tsx]: see if you can move `text-sm font-normal` from 44 and 45 to 43.
And check if it works.

File: src/components/common/searchbar.tsx [hunk 3]
```diff
@@ -21,21 +23,28 @@ const SearchBar: FC<Props> = ({
...
-      />
+      <span className="search-input p-input-icon-left hidden sm:block">
+        <CustomImage
+          src={HeaderSearch}
+          alt="Search"
+          className="absolute left-5 bottom-3"
+        />
+        <CustomInputText
+          type="text"
+          className="w-full min-w-[400px] h-[40px] !rounded-full focus:outline-none  !bg-[#30343B] placeholder:text-[#FAF9F6]placeholder:font-helvetica placeholder:font-normal !shadow-none !border-none"
```
Comment (raj-kumar-litmus) [inline src/components/common/searchbar.tsx]: There is no space in between `placeholder:text-[#FAF9F6]placeholder:font-helvetica` 
Check if style is applied as expected

File: src/components/orderstatuspopup.tsx [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+  description: string;
+}
+
+const OrderStatusPopup = () => {
+  const HEADERS = [
+    { field: "code", header: "Status Code" },
+    { field: "description", header: "Description" },
+  ];
+
+  const [orderStatusList, setOrderStatusList] = useState<Array<OrderStatus>>(
```
Comment (raj-kumar-litmus) [inline src/components/orderstatuspopup.tsx]: `useState<OrderStatus[]>`

File: src/components/orderstatuspopup.tsx [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+    []
+  );
+  const [openDialog, setOpenDialog] = useState<boolean>(false);
+
+  useEffect(() => {
```
Comment (raj-kumar-litmus) [inline src/components/orderstatuspopup.tsx:23]: useEffect(() => {
    setOrderStatusList(ORDER_STATUS_LIST);
  }, [ORDER_STATUS_LIST]);

File: src/components/orderstatuspopup.tsx [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+    setOrderStatusList(ORDER_STATUS_LIST);
+  }, []);
+
+  return (
+    <div>
```
Comment (raj-kumar-litmus) [inline src/components/orderstatuspopup.tsx]: use `<>` instead of `div`

File: src/components/promotionspopup.tsx [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+import promotionsJSON from "../sampleJSON/promotions.json";
+import { IPromotion } from "../@types/promotion";
+import CustomDialog from "./common/customdialog";
+import PromotionCard from "./common/promotioncard";
+import CustomButton from "./Button";
+import { fetchData } from "../utils/fetchUtil";
+import { URL_PROMOTIONS } from "../constants/apiConstants";
+
+const PromotionsPopup = () => {
+  const [promotions, setPromotions] = useState<Array<IPromotion>>([]);
```
Comment (raj-kumar-litmus) [inline src/components/promotionspopup.tsx]: same comment as mentioned earlier.
` useState<IPromotion[]>([]);`

File: src/components/promotionspopup.tsx [hunk 8]
```diff
@@ -0,0 +1,56 @@
...
+  const [orderId, setOrderId] = useState<string>("60577546279");
+  const [openDialog, setOpenDialog] = useState<boolean>(false);
+
+  const getPromotions = async () => {
+    const params = {
+      orderId: orderId,
+    };
+    try {
+      const data = await fetchData(URL_PROMOTIONS, params);
+      // setPromotions(data || []);
```
Comment (raj-kumar-litmus) [inline src/components/promotionspopup.tsx:21]: remove commented lines.

File: src/components/promotionspopup.tsx [hunk 9]
```diff
@@ -0,0 +1,56 @@
...
+      console.error("Error while fetching data:", err);
+    }
+  };
+
+  useEffect(() => {
+    getPromotions();
+  }, []);
+
+  return (
+    <div>
```
Comment (raj-kumar-litmus) [inline src/components/promotionspopup.tsx]: use `<>`

File: src/components/orderstatuspopup.tsx (hunk 5 above)
Comment (thasnim-cm) [reply src/components/orderstatuspopup.tsx:23]: Is this required? ORDER_STATUS_LIST  is static

File: src/components/common/promotioncard.tsx (hunk 2 above)
Comment (thasnim-cm) [reply src/components/common/promotioncard.tsx]: It worked. Updated the code

File: src/components/common/searchbar.tsx (hunk 3 above)
Comment (thasnim-cm) [reply src/components/common/searchbar.tsx]: Updated the code

File: src/components/orderstatuspopup.tsx (hunk 4 above)
Comment (thasnim-cm) [reply src/components/orderstatuspopup.tsx]: Done

File: src/components/orderstatuspopup.tsx (hunk 6 above)
Comment (thasnim-cm) [reply src/components/orderstatuspopup.tsx]: Done

File: src/components/promotionspopup.tsx (hunk 7 above)
Comment (thasnim-cm) [reply src/components/promotionspopup.tsx]: Done

File: src/components/promotionspopup.tsx (hunk 9 above)
Comment (thasnim-cm) [reply src/components/promotionspopup.tsx]: Done

File: tsconfig.node.json [hunk 10]
```diff
@@ -1,6 +1,6 @@
 {
   "compilerOptions": {
-    "composite": true,
```
Comment (raj-kumar-litmus) [inline tsconfig.node.json]: why did u comment this out ?

Comment (thasnim-cm) [reply tsconfig.node.json]: Uncommented and updated MR

File: src/components/orderstatuspopup.tsx (hunk 5 above)
Comment (raj-kumar-litmus) [reply src/components/orderstatuspopup.tsx:23]: It's a good practise to include the dependencies.

File: src/components/promotionspopup.tsx [hunk 11]
```diff
@@ -18,7 +18,6 @@ const PromotionsPopup = () => {
     };
     try {
       const data = await fetchData(URL_PROMOTIONS, params);
```
Comment (raj-kumar-litmus) [inline src/components/promotionspopup.tsx:20]: are u not getting any eslint warning for `data` being unused ?

Comment (thasnim-cm) [reply src/components/promotionspopup.tsx:20]: It shows that the 'data' variable is declared, but its value is never read. We might need this variable in the future when integrating the API, which is why I've kept it there

File: src/components/orderstatuspopup.tsx (hunk 5 above)
Comment (thasnim-cm) [reply src/components/orderstatuspopup.tsx:23]: Updated the  MR

**Pull request merged**